### PR TITLE
fix(helm): pass down username/password CLI parameters to OCI registry clients (v3 backport)

### PR DESCRIPTION
**Pull request by banjoh**
**What this PR does / why we need it**:

Backport https://github.com/helm/helm/pull/12769 to v3 branch

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by joejulian — APPROVED** [review]

File: cmd/helm/push.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ type registryPushOptions struct {
 	caFile                string
 	insecureSkipTLSverify bool
 	plainHTTP             bool
+	password              string
+	username              string
```
Comment (TerryHowe) [inline cmd/helm/push.go:44]: Too late now, but I thought the idea was to authenticate before using push

**Review by sabre1041 — APPROVED** [review]
LGTM

File: cmd/helm/push.go (hunk 1 above)
Comment (banjoh) [reply cmd/helm/push.go:44]: > authenticate before using push

Do you mean `helm registry login` then `helm push` or are you referring to a logic in code?

Comment (TerryHowe) [reply cmd/helm/push.go:44]: Yeh, I thought these commands were set up to authenticate with `helm login` and all the other commands need to deal with TLS etc, but not authn Sounds like my assumption was incorrect.

Comment (sabre1041) [reply cmd/helm/push.go:44]: > Yeh, I thought these commands were set up to authenticate with `helm login` and all the other commands need to deal with TLS etc, but not authn Sounds like my assumption was incorrect.

You can do both. You can login to allow multiple invocations without having the explicitly specifying each option, or one can integrate ad-hoc, as enabled through this enhancement

**Comment by EronWright** [comment]
Should this be assigned to the [3.17.0](https://github.com/helm/helm/milestone/147) milestone?

**Comment by mardonner** [comment]
Is this v3 backport currently part of any release?
With Helm 3.18.4 I get 401s when using --username and --password.
Only helm registry login works for me.

**Comment by banjoh** [comment]
> Is this v3 backport currently part of any release?

Yes, this is part of v3

> With Helm 3.18.4 I get 401s when using --username and --password

It was released in [v3.17.0](https://github.com/helm/helm/releases/tag/v3.17.0) and it works with that version. It doesn't in v3.18.4. I'll investigate

**Pull request merged**